pkg/volume/util/subpath/subpath_windows.go

```diff
@@ -201,6 +201,12 @@ func lockAndCheckSubPathWithoutSymlink(volumePath, subPath string) ([]uintptr, e
 			break
 		}
 
+		// go1.23 behavior change: https://github.com/golang/go/issues/63703#issuecomment-2535941458
+		if stat.Mode()&os.ModeIrregular != 0 {
+			errorResult = fmt.Errorf("subpath %q is an unexpected irregular file after EvalSymlinks", currentFullPath)
```

Comment:
is this error correct prior to dropping the winsymlink compat switch workaround, or should we be dropping that workaround in this PR to make this code correct?